### PR TITLE
limits test: Try finding new limits, record wallclock and explain wallclock times

### PR DESCRIPTION
**Pull request by def-**
Fixes: https://github.com/MaterializeInc/database-issues/issues/8658
### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

**Comment by def-** [comment]
This requires https://github.com/MaterializeInc/materialize/pull/30109 to be merged first, so only review the second commit.

File: ci/release-qualification/pipeline.template.yml [hunk 1]
```diff
@@ -473,3 +473,16 @@ steps:
...
+    depends_on: build-aarch64
+    agents:
+      queue: hetzner-aarch64-16cpu-32gb
+    plugins:
+      - ./ci/plugins/mzcompose:
+          composition: limits
+          run: main
+          args: [--find-limit]
+    timeout_in_minutes: 3600
+    parallelism: 20
```
Comment (nrainer-materialize) [inline ci/release-qualification/pipeline.template.yml:488]: Sounds a bit expensive but ok. :)

**Review by nrainer-materialize — APPROVED** [review]

**Comment by nrainer-materialize** [comment]
Have you encountered any issues that build jobs get stuck when exceeding limits like I did in bounded-memory?

**Comment by nrainer-materialize** [comment]
> Yes, they do get extremely long-running, I marked those scenarios with `FIND_NEW_LIMITS = False`.

Ok, interesting. That means it depends on the scenario.

File: misc/python/materialize/test_analytics/setup/tables/13-product-limits.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
+-- Copyright Materialize, Inc. and contributors. All rights reserved.
```
[the remark this answers is not in the document]
Comment (def-) [reply misc/python/materialize/test_analytics/setup/tables/13-product-limits.sql:1]: Too much progress, how will 10 digits be enough? ;)

**Pull request merged**